### PR TITLE
Small refactorings/tweaks to sentry

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Small refactorings to Sentry. Doesn't change any behavior, but includes some clean-ups, in particular:

- Clarifies what `NoDefaultTokenAudience` means, since that never shipped as an actual feature flag. Changed TODOs to say that the code should be removed in Dapr 1.12 instead (it's only needed for smooth upgrades from Dapr 1.9 and before)
- When comparing crypto keys for equality, use the `.Equal` method from the stdlib rather than our own
- Ensure logs start with an uppercase letter, unlike errors
- Improve the wording and/or format for some error messages and logs

### CODE REVIEW

**Review by pravinpushkar — DISMISSED** [review]

File: pkg/sentry/identity/kubernetes/validator.go [hunk 1]
```diff
@@ -19,21 +19,19 @@ const errPrefix = "csr validation failed"
...
 	}
 }
 
 type validator struct {
 	client    k8s.Interface
 	auth      kauth.AuthenticationV1Interface
 	audiences []string
-	// TODO: Remove once the NoDefaultTokenAudience feature is finalized
+	// TODO: Remove in Dapr 1.12 to enforce setting an explicit audience
 	noDefaultTokenAudience bool
```
Comment (DeepanshuA) [inline pkg/sentry/identity/kubernetes/validator.go:35]: Should we remove it from here and simply set `true` in https://github.com/dapr/dapr/blob/d1a4c7945da5cb0087cfaac8531662156e84bee8/pkg/sentry/identity/kubernetes/validator.go#L57 ?

Comment (ItalyPaleAle) [reply pkg/sentry/identity/kubernetes/validator.go]: Done

**Review by DeepanshuA — APPROVED** [review]

**Review by mukundansundar — COMMENTED** [review]
overall looks good. Had a couple of questions.

File: pkg/sentry/certs/certs.go [hunk 2]
```diff
@@ -21,13 +22,13 @@ const (
 
 // Credentials holds a certificate and private key.
 type Credentials struct {
-	PrivateKey  interface{}
+	PrivateKey  crypto.PrivateKey
```
[the remark this answers is not in the document]
Comment (ItalyPaleAle) [reply pkg/sentry/certs/certs.go:25]: If this breaks anything, then we have an error that must be caught during development.

File: pkg/sentry/sentry.go [hunk 3]
```diff
@@ -181,10 +181,7 @@ func (s *sentry) createValidator() (identity.Validator, error) {
 			return nil, fmt.Errorf("failed to create kubernetes client: %w", err)
 		}
 
-		// TODO: Remove once the NoDefaultTokenAudience feature is finalized
-		noDefaultTokenAudience := false
-
-		return kubernetes.NewValidator(kubeClient, s.conf.GetTokenAudiences(), noDefaultTokenAudience), nil
+		return kubernetes.NewValidator(kubeClient, s.conf.GetTokenAudiences()), nil
```
[the remark this answers is not in the document]
Comment (ItalyPaleAle) [reply pkg/sentry/sentry.go:184]: it appears so, yes

**Review by mukundansundar — APPROVED** [review]

**Comment by daixiang0** [comment]
> Ensure logs start with an uppercase letter, unlike errors

Is it a rule for all contributors?

**Pull request merged**